exercises/practice/zebra-puzzle/zebra_puzzle.cpp

```diff
@@ -0,0 +1,21 @@
+#include "zebra_puzzle.h"
+
+#include <string>
+
+using namespace std;
+
+namespace zebra_puzzle 
+{
+
+typedef struct 
+{
+    std::string drinksWater;
+    std::string ownsZebra;
+} Solution;
```

Comment:
*\*If\** you want to define that type for our students, do that in the `.h` file where it is required.  
(and without the `typedef` idiom, see above.)